UnitsNet.Tests/CustomCode/ScalarTests.cs

```diff
@@ -0,0 +1,68 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by \generate-code.bat.
+//
+//     Changes to this file will be lost when the code is regenerated.
+//     The build server regenerates the code before each build and a pre-build
+//     step will regenerate the code on each local build.
+//
+//     See https://github.com/angularsen/UnitsNet/wiki/Adding-a-New-Unit for how to add or edit units.
+//
+//     Add CustomCode\Quantities\MyQuantity.extra.cs files to add code to generated quantities.
+//     Add UnitDefinitions\MyQuantity.json and run generate-code.bat to generate new units or quantities.
+//
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+// Licensed under MIT No Attribution, see LICENSE file at the root.
+// Copyright 2013 Andreas Gullberg Larsen (andreas.larsen84@gmail.com). Maintained at https://github.com/angularsen/UnitsNet.
+
+using System;
+using Xunit;
+
+namespace UnitsNet.Tests.CustomCode
+{
+    public class ScalarTests : ScalarTestsBase
+    {
+        // Override properties in base class here
+        protected override double AmountInOneAmount => 1;
+        protected override bool SupportsSIUnitSystem => false;
+
+        /// <summary>
+        /// Purposely did not call it an Int as it is not an int underneath.
+        /// </summary>
+        [Fact]
+        public void ScalarWholeNumberEqualsScalarWholeNumber()
```

Comment:
I love tests, but I believe these test cases are already covered by the generated tests:

- ArithmeticOperators
- ConversionRoundTrip

https://github.com/angularsen/UnitsNet/pull/889/files#diff-136cd525eae5af5b9f86b098cdab2bc0741d8119d188b00e60524aada8d407a5R184-R202

Reply:
I did not see those, the one I did see was 
`Assertion(3, ScalarUnit.Amount, Quantity.From(3, ScalarUnit.Amount));` from IQuantityTests.g.cs

Looking from someone not familiar with the codebase one thing that I will add about the tests that you listed (ArithmeticOperators & ConversionRoundTrip) - they only use integers to test. Looking at the code underneath, it will not make a difference as they are stored as a double anyway, but if that implementation ever changes there may not be sufficient test coverage. I think this is outside of the scope of this PR, so I will remove those tests.